js/app.js

```diff
@@ -0,0 +1,123 @@
+'use strict';
+
+function getResources() {
+  // if calling consecutive times remove all children generated by prior request
+  while (contentContainer.firstChild) {
+    contentContainer.removeChild(contentContainer.firstChild);
+  }
+
+  const swapi = 'https://swapi.co/api/';
+  let attr1 = document.createElement('h2');
+  let attr2 = document.createElement('p');
+  let attr3 = document.createElement('p');
+  contentContainer.appendChild(attr1);
+  contentContainer.appendChild(attr2);
+  contentContainer.appendChild(attr3);
+
+  if (resourceType.value === 'people') {
+    function peopleXmlListener() {
+      if (this.status < 200 || this.status >= 300) {
+        let error = document.createElement('h2');
```

Comment:
You could turn the contents of this if body into a function and just call it from the if. You could also return after calling the function so you wouldn't need the else below.

Reply:
I did this, but because of hoisting it threw a fit.. I accommodated that, but am not sure if I did so in a "best practice" way

Reply:
What you have is fine, but since you're just using the status and responseText from the context, you could just pass in those two things rather than the whole context.